layers/gpu_validation/gpu_validation.cpp

```diff
@@ -1360,150 +1384,81 @@ void GpuAssisted::UpdateBoundDescriptors(VkCommandBuffer commandBuffer, VkPipeli
-                    auto desc = s.bound_descriptor_set;
-                    if (desc && (desc->GetBindingCount() > 0)) {
-                        auto layout = desc->GetLayout();
-                        // For each set, fill in index of its bindings sizes in the sizes array
-                        *sets_to_sizes++ = bind_counter;
-                        // For each set, fill in the index of its bindings in the bindings_to_written array
-                        *sets_to_bindings++ = bind_counter + number_of_sets + binding_count;
-                        for (auto &binding : *desc) {
-                            // For each binding, fill in its size in the sizes array
-                            // Shader instrumentation is tracking inline uniform blocks as scalers. Don't try to validate inline
-                            // uniform blocks
-                            if (VK_DESCRIPTOR_TYPE_INLINE_UNIFORM_BLOCK_EXT == binding->type) {
-                                sizes[binding->binding] = 1;
-                            } else {
-                                sizes[binding->binding] = binding->count;
-                            }
-                            // Fill in the starting index for this binding in the written array in the bindings_to_written array
-                            bindings_to_written[binding->binding] = written_index;
-
-                            // Shader instrumentation is tracking inline uniform blocks as scalers. Don't try to validate inline
-                            // uniform blocks
-                            if (VK_DESCRIPTOR_TYPE_INLINE_UNIFORM_BLOCK_EXT == binding->type) {
-                                data_ptr[written_index++] = UINT_MAX;
-                                continue;
-                            }
-
-                            if ((binding->binding_flags & VK_DESCRIPTOR_BINDING_UPDATE_AFTER_BIND_BIT) != 0) {
-                                di_input_block.update_at_submit[written_index] = binding.get();
-                            } else {
-                                SetBindingState(data_ptr, written_index, binding.get());
-                            }
-                            written_index += binding->count;
-                        }
-                        auto last = desc->GetLayout()->GetMaxBinding();
-                        bindings_to_written += last + 1;
-                        bind_counter += last + 1;
-                        sizes += last + 1;
-                    } else {
-                        *sets_to_sizes++ = 0;
-                        *sets_to_bindings++ = 0;
+
+            // For each set, allocate an input buffer that describes the descriptor set and its update status as follows
+            // Word 0 = the number of bindings in the descriptor set - note that the bindings can be sparse and this is the largest
+            // binding number + 1 which we'll refer to as N Words 1 through Word N  = the number of descriptors in each binding
```

Comment:
Having some difficulty parsing this. Maybe some additional punctuation?
```suggestion
            // binding number + 1 which we'll refer to as N. Words 1 through Word N = the number of descriptors in each binding.
```